controllers/story-controller.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file Controller RESTful Web service API for stories resource
+ */
+import StoryDao from "../daos/story-dao";
+import StoryControllerI from "../interfaces/StoryControllerI";
+import {Express, Request, Response} from "express";
+import Story from "../models/stories/story";
+
+/**
+ * @class StoryController Implements RESTful Web service API for stories resource.
+ * Defines the following HTTP endpoints:
+ * <ul>
+ *     <li>POST /api/users/:uid/stories to create a new story instance for a given user</li>
+ *     <li>DELETE /api/stories/:sid to remove a particular story instance</li>
+ *     <li>DELETE /api/users/:uid/stories to remove all stories of the particular user</li>
+ *     <li>GET /api/stories/:sid to retrieve a particular story instances</li>
+ *     <li>GET /api/users/:uid/stories/:sid to retrieve stories for a given user </li>
+ *     <li>GET /api/stories to retrieve all the story instances</li>
+ * </ul>
+ * @property {TuitDao} tuitDao Singleton DAO implementing tuit CRUD operations
+ * @property {TuitController} tuitController Singleton controller implementing
+ * RESTful Web service API
+ */
+export default class StoryController implements StoryControllerI {
+  private static storyDao: StoryDao = StoryDao.getInstance();
+  private static storyController: StoryController | null = null;
+
+  /**
+   * Creates singleton controller instance
+   * @param {Express} app Express instance to declare the RESTful Web service API
+   * @return StoryController
+   */
+  public static getInstance = (app: Express): StoryController => {
+    if(StoryController.storyController === null) {
+      StoryController.storyController = new StoryController();
+
+      app.post("/api/users/:uid/stories", StoryController.storyController.createStory);
+      app.delete("/api/stories/:sid", StoryController.storyController.deleteStoryByID);
+      app.delete("/api/users/:uid/stories", StoryController.storyController.deleteAllStories);
+      app.get("/api/stories/:sid", StoryController.storyController.findStoryById);
+      app.get("/api/users/:uid/stories/:sid", StoryController.storyController.findStoriesByUser);
+      app.get("/api/stories", StoryController.storyController.findStories);
+    }
+    return StoryController.storyController;
+  }
+
+  private constructor() {}
+
+  /**
+   * Creates a new story instance
+   * @param {Request} req Represents request from client, including body
+   * containing the JSON object for the new story to be inserted in the
+   * database
+   * @param {Response} res Represents response to client, including the
+   * body formatted as JSON containing the new story that was inserted in the
+   * database
+   */
+  createStory = (req: Request, res: Response) =>
```

Comment:
Looks good! One thing: I think we can add the session id for user for all user related methods.

Reply:
Thank you! 
I couldn't reply to your previous comment.
1. Renamed interfaces
2. Renamed deleteAllStories as you suggested
3. Added session id

Reply:
Awesome, thanks!